Tests/SKCoreTests/BuildServerBuildSystemTests.swift

```diff
@@ -65,6 +66,41 @@ final class BuildServerBuildSystemTests: XCTestCase {
     }
   }
 
+  func testBuildTargets() {
+    let root = AbsolutePath(
+      inputsDirectory().appendingPathComponent(testDirectoryName, isDirectory: true).path)
+    let buildFolder = AbsolutePath(NSTemporaryDirectory())
+    let buildSystem = try? BuildServerBuildSystem(projectRoot: root, buildFolder: buildFolder)
+    XCTAssertNotNil(buildSystem)
+
+    let expectation = XCTestExpectation(description: "build target expectation")
+
+    buildSystem?.buildTargets(reply: { response in
+      switch(response) {
+      case .success(let targets):
+        XCTAssertEqual(targets, [
+                       BuildTarget(id: BuildTargetIdentifier(uri: URL(string: "first_target")!),
+                                   displayName: "First Target",
+                                   baseDirectory: URL(fileURLWithPath: "/some/dir"),
+                                   tags: [BuildTargetTag.library, BuildTargetTag.test],
+                                   capabilities: BuildTargetCapabilities(canCompile: true, canTest: true, canRun: false),
+                                   languageIds: [Language.objective_c, Language.swift],
+                                   dependencies: []),
+                       BuildTarget(id: BuildTargetIdentifier(uri: URL(string: "second_target")!),
+                                   displayName: "Second Target",
+                                   baseDirectory: URL(fileURLWithPath: "/some/dir"),
+                                   tags: [BuildTargetTag.library, BuildTargetTag.test],
+                                   capabilities: BuildTargetCapabilities(canCompile: true, canTest: false, canRun: false),
+                                   languageIds: [Language.objective_c, Language.swift],
+                                   dependencies: [BuildTargetIdentifier(uri: URL(string: "first_target")!)]),
+                       ])
+        expectation.fulfill()
+      case .failure(let error):
+        XCTFail(error.message)
+      }
+    })
+    XCTAssertEqual(XCTWaiter.wait(for: [expectation], timeout: 1), .completed)
```

Comment:
Since this is a positive test (ie. we expect it to not timeout), maybe make this a bit longer - when running in CI things can sometimes be heavily loaded.  For tests that expect to timeout like `testSwiftDocumentBuildSettingsChangedFalseAlarm`, a shorter value is fine.

Reply:
Went with 15s like the other tests